app/controllers/games_controller.rb

```diff
@@ -13,6 +13,16 @@ def create
 
   def show
     @game = Game.find_by_id(params[:id])
+
+    data = params[:p_type]
+
+    until data.nil?
+      @game.pieces.where(type: 'Pawn', vertical_position: 1).first.try(:update_attributes, type: data)
+      @game.pieces.where(type: 'Pawn', vertical_position: 8).first.try(:update_attributes, type: data)
+      data = params[:p_type] = nil
```

Comment:
Not anymore, it looks like. With the way I had the code set up before, the first pawn promotion would cause any subsequent eight rank pawns to automatically update to the whatever was selected